gorma.go

```diff
@@ -120,19 +184,185 @@ func (g *Generator) Generate(api *design.APIDefinition) ([]string, error) {
+			actionable := false
+			err = res.IterateActions(func(ad *design.ActionDefinition) error {
+				if hasUserType(ad) {
+					actionable = true
+				}
+				return nil
+
+			})
+			if !actionable {
+				return nil
+			}
+
+			if !res.SupportsVersion(v.Version) {
+				return nil
+			}
+			prefix := "resource"
+			if v.Version != "" {
+				prefix = prefix + "v" + codegen.Goify(v.Version, false)
+
+			}
+			name := strings.ToLower(res.Name)
+
+			mediafilename := filepath.Join(modelDir(), name+prefix+"_genmodel.go")
+			os.Remove(mediafilename)
+
+			resw, err := NewResourceWriter(mediafilename)
+			if err != nil {
+				fmt.Println("Error executing Gorma: ", err.Error())
+				panic(err)
+			}
+			resw.WriteHeader(title, "models", imports)
+
+			err = resw.Execute(v.Version, res)
+			if err != nil {
+				fmt.Println("Error executing Gorma: ", err.Error())
+				g.Cleanup()
+				return err
+			}
+			if err := resw.FormatCode(); err != nil {
+				fmt.Println("Error executing Gorma: ", err.Error())
+				g.Cleanup()
+				return err
+			}
+			if err == nil {
+				g.genfiles = append(g.genfiles, mediafilename)
+			}
+			return nil
+		})
+		return nil
+	})
+	return err
+}
+
+// Generate produces the generated media files
+func (g *Generator) generateMedia(api *design.APIDefinition) error {
+	os.MkdirAll(modelDir(), 0755)
+	app := kingpin.New("Model generator", "model generator")
+	codegen.RegisterFlags(app)
+	_, err := app.Parse(os.Args[1:])
+	if err != nil {
+		panic(err)
+	}
+	gopath := filepath.SplitList(os.Getenv("GOPATH"))[0]
+
+	imp, err := filepath.Rel(filepath.Join(gopath, "src"), codegen.OutputDir)
+	if err != nil {
+		return err
+	}
+	imp = path.Join(filepath.ToSlash(imp), "app")
+	imports := []*codegen.ImportSpec{
+		codegen.SimpleImport(imp),
+		codegen.SimpleImport("github.com/jinzhu/copier"),
+	}
+	// get the imports for the app packages
+	api.IterateVersions(func(v *design.APIVersionDefinition) error {
+		if v.IsDefault() {
+			return nil
+		}
+		imports = append(imports, codegen.SimpleImport(imp+"/"+codegen.Goify(codegen.VersionPackage(v.Version), false)))
+		return nil
+	})
+
+	title := fmt.Sprintf("%s: Media Helpers", api.Name)
+
+	err = api.IterateVersions(func(v *design.APIVersionDefinition) error {
+		err = v.IterateMediaTypes(func(res *design.MediaTypeDefinition) error {
+			if res.Reference == nil {
+				return nil
+			}
+			if res.Type.IsObject() {
+				if !res.SupportsVersion(v.Version) {
+					return nil
+				}
+				prefix := "media"
+				if v.Version != "" {
+					prefix = prefix + "v" + codegen.Goify(v.Version, false)
+
+				}
+				name := strings.ToLower(res.TypeName)
+
+				mediafilename := filepath.Join(modelDir(), name+prefix+"_genmodel.go")
+
+				os.Remove(mediafilename)
+				resw, err := NewMediaWriter(mediafilename)
+				if err != nil {
+					fmt.Println("Error executing Gorma: ", err.Error())
+					panic(err)
+				}
+				resw.WriteHeader(title, "models", imports)
+
+				err = resw.Execute(v.Version, res)
```

Comment:
Need to check for gorma Models here:

```go
if md, ok := metaLookup(res.Metadata, ""); ok && md == "Model" {
```
This will currently gen code for models that are not labeled as gorma Models.